sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2769,111 +2795,145 @@ public <T> Mono<FeedResponse<T>> readMany(
+                                                   ModelBridgeInternal.toObjectFromJsonSerializable(document, klass)).collect(Collectors.toList()));
+                                               aggregateRequestStatistics.addAll(diagnosticsAccessor.getClientSideRequestStatistics(page.getCosmosDiagnostics()));
+                                           }
+
+                                           // NOTE: This CosmosDiagnostics instance intentionally isn't captured in the
+                                           // ScopedDiagnosticsFactory - and a ssuch won't be included in the diagnostics of the
+                                           // CosmosDiagnosticsContext - which is fine, because the CosmosDiagnosticsContext
+                                           // contains the "real" CosmosDiagnostics instances (which will also be used
+                                           // for diagnostics purposes - like metrics, logging etc.
+                                           // this artificial CosmosDiagnostics with the aggregated RU/s etc. si simply
+                                           // to maintain the API contract that a FeedResponse returns one CosmosDiagnostics
+                                           CosmosDiagnostics aggregatedDiagnostics = BridgeInternal.createCosmosDiagnostics(aggregatedQueryMetrics);
+                                           diagnosticsAccessor.addClientSideDiagnosticsToFeed(
+                                               aggregatedDiagnostics, aggregateRequestStatistics);
+
+                                           state.mergeDiagnosticsContext();
+                                           diagnosticsAccessor
+                                               .setDiagnosticsContext(
+                                                   aggregatedDiagnostics,
+                                                   state.getDiagnosticsContextSnapshot());
+
+                                           headers.put(HttpConstants.HttpHeaders.REQUEST_CHARGE, Double
+                                               .toString(requestCharge));
+                                           FeedResponse<T> frp = BridgeInternal
+                                               .createFeedResponseWithQueryMetrics(
+                                                   finalList,
+                                                   headers,
+                                                   aggregatedQueryMetrics,
+                                                   null,
+                                                   false,
+                                                   false,
+                                                   aggregatedDiagnostics);
+                                           return frp;
+                                       });
+                        })
+                        .onErrorMap(throwable -> {
```

Comment:
Is this a catch all case of sorts?

Reply:
Just a way to transform exceptions in Reactor

Reply:
In our case we don't handle the exceptions - you could thought also map an exception to a success value - but we just modify the exception here.